src/adapters/Cornerstone/Segmentation_4X.js

```diff
@@ -57,7 +57,8 @@ function generateSegmentationFromDatasets(
     inputLabelmaps3D,
     userOptions = {}
 ) {
-    //on multiframe images, datasets array contains only 1 element but should work as well
+    //make sure _meta property is present in every element of datasets with at least an empty array
+    datasets = datasets.map(ds => ({ ...ds, _meta: ds._meta || [] }));
```

Comment:
Hi - thanks for this, I'm trying to get my head back into the details.

Wouldn't we get the same effect more generally if [this line](https://github.com/dcmjs-org/dcmjs/blob/master/src/derivations/DerivedDataset.js#L41) where changed to:

`_meta: this.referencedDataset._meta || []`

Reply:
Hi Pieper,

Yes, much better. Thanks